accounts-db/src/tiered_storage/footer.rs

```diff
@@ -182,14 +209,19 @@ impl TieredStorageFooter {
 
     pub fn new_from_footer_block(file: &TieredStorageFile) -> TieredStorageResult<Self> {
         let mut footer_size: u64 = 0;
-        let mut footer_version: u64 = 0;
-        let mut magic_number = TieredStorageMagicNumber(0);
-
         file.seek_from_end(-(FOOTER_TAIL_SIZE as i64))?;
         file.read_type(&mut footer_size)?;
+        if footer_size != FOOTER_SIZE as u64 {
```

Comment:
I think we need to first check the footer version here before the sanization.

If the version matches the current version, then we run the following sanitization check for the current version.

Otherwise, we run the sanitization check for that specific footer version.